draft-ietf-tls-esni.md

```diff
@@ -274,25 +280,32 @@ constrain server name lengths. Names may exceed this length if, e.g.,
 the server uses wildcard names or added new names to the anonymity set.
 
 public_name
-: The non-empty name of the client-facing server, i.e., the entity trusted to
-update the ECH configuration. This is used to correct misconfigured clients, as
-described in {{handle-server-response}}. This value MUST NOT begin or end with
-an ASCII dot and MUST be parsable as a dot-separated sequence of LDH labels, as
-defined in {{!RFC5890, Section 2.3.1}}. Clients MUST ignore any `ECHConfig`
-structure whose `public_name` does not meet these criteria. Note that these
-criteria are incomplete; they incidentally rule out textual representations of
-IPv6 addresses (see {{!RFC3986, Section 3.2.2}}), but do not exclude IPv4
-addresses in standard dotted-decimal or other non-standard notations such as
-octal and hexadecimal (see {{RFC3986, Section 7.4}}). If `public_name` contains
-a literal IPv4 or IPv6 address, the client SHOULD ignore the `ECHConfig` to
-avoid sending a non-compliant "server_name" extension on the ClientHelloOuter
-(see {{!RFC6066, Section 3}}).
+: The DNS name of the client-facing server, i.e., the entity trusted
+to update the ECH configuration. This is used to correct misconfigured clients,
+as described in {{handle-server-response}}.
+
+: This value MUST NOT begin or end with an ASCII dot and MUST be parsable as a
+dot-separated sequence of LDH labels, as defined in
+{{!RFC5890, Section 2.3.1}}. Clients MUST ignore any `ECHConfig` structure
+whose public_name does not meet these criteria.
+
```

Comment:
```suggestion
: Clients MUST ignore any `ECHConfig` structure
whose public_name does is not parsable as a
dot-separated sequence of LDH labels, as defined in
{{!RFC5890, Section 2.3.1}} or which
begins or end with an ASCII dot.
```

Reply:
Note that I think this still allows for bogus names, like ones that are super-long. Do we want to allow clients to check for that?

Reply:
The public_name field is already limited to 255 characters, so I think we're fine.